Core/src/org/sleuthkit/autopsy/modules/plaso/PlasoIngestModule.java

```diff
@@ -0,0 +1,410 @@
+import org.sleuthkit.datamodel.SleuthkitCase;
+
+/**
+ * Data source ingest module that runs plaso against the image 
+ */
+
+public class PlasoIngestModule implements DataSourceIngestModule {
+
+    private static final Logger logger = Logger.getLogger(PlasoIngestModule.class.getName());
+    private static final IngestServices services = IngestServices.getInstance();
+    private static final String PLASO = "plaso";
+    private static final String PLASO64 = "plaso//plaso-20180127-amd64";
+    private static final String PLASO32 = "plaso//plaso-20180127-win32";
+    private static final String LOG2TIMELINE_EXECUTABLE = "Log2timeline.exe"; 
+    private static final String PSORT_EXECUTABLE = "psort.exe";
+    private static final String VSS_OPTIONS = "--vss-stores";
+    private static final String PARTITIONS = "--partitions";
+    private static final String HASHER_FILE_SIZE_LIMIT = "--hasher_file_size_limit";
+    private static final String ONE = "1";
+    private static final String HASHERS = "--hashers";
+    private static final String NONE = "none";
+    private static final String ALL = "all";
+    private IngestJobContext context;
+
+    PlasoIngestModule() {
+    }
+
+    @Override
+    public void startUp(IngestJobContext context) throws IngestModuleException {
+        this.context = context;
+
+    }
+
+    @NbBundle.Messages({
+        "PlasoIngestModule_startUp_message=Starting Plaso Run.",
+        "PlasoIngestModule_error_running=Error running Plaso."
+    })
+
+    @Override
+    public ProcessResult process(Content dataSource, DataSourceIngestModuleProgress statusHelper) {
+        String imgName = dataSource.getName();
+        File log2TimeLineExecutable;
+        File psortExecutable;
+        Case currentCase;
+        String moduleOutputPath;
+        String tempOutputPath;
+        
+        Image img = (Image) dataSource;
+
+        currentCase = Case.getCurrentCase();
+
+        log2TimeLineExecutable = locateExecutable(LOG2TIMELINE_EXECUTABLE);
+        if (log2TimeLineExecutable == null) {
+            //TODO   Throw exeption
+        }
+        psortExecutable = locateExecutable(PSORT_EXECUTABLE);
+        if (psortExecutable == null) {
+            //TODO   Throw exeption
+        }
+        
+        String[] imgFile = img.getPaths();
+        
+        /*
+        * Make an module and temp output folder unique to this data source.
+        */
+        Long dataSourceId = dataSource.getId();
+        moduleOutputPath = Paths.get(currentCase.getModuleDirectory(), PLASO).toString();
+        File directory = new File(String.valueOf(moduleOutputPath));
+        if (!directory.exists()) {
+            directory.mkdirs();   
+        }
+        tempOutputPath = Paths.get(currentCase.getTempDirectory(), PLASO, imgName).toString();
+        directory = new File(String.valueOf(tempOutputPath));
+        if (!directory.exists()) {
+            directory.mkdirs();
+        }
```

Comment:
The preceding directory creation code looks incomplete. The data source object id is queried, but not used. However, although data source object ids are unique within a case, it is possible that ingest is run multiple times on the same data source, so it is safest to do what we talked about in stand up, i.e., to use a time stamp from TimeStampUtils.createTimeStamp instead (in combination with a data source if you want to be really safe) as a way to get a unique path. In any case, a unique suffix is not actually being used here for the module output directory, and the temp directory will not be guaranteed unique because image names are not guaranteed to be unique.